internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -146,23 +146,24 @@ func InvokeWatcher(log *logger.Logger, agentExecutable string) (*exec.Cmd, error
 	}
 
 	cmd := invokeCmd(agentExecutable)
-	defer func() {
-		if cmd.Process != nil {
-			log.Infof("releasing watcher %v", cmd.Process.Pid)
-			_ = cmd.Process.Release()
-		}
-	}()
-
 	log.Infow("Starting upgrade watcher", "path", cmd.Path, "args", cmd.Args, "env", cmd.Env, "dir", cmd.Dir)
 	if err := cmd.Start(); err != nil {
 		return nil, fmt.Errorf("failed to start Upgrade Watcher: %w", err)
 	}
 
 	upgradeWatcherPID := cmd.Process.Pid
 	agentPID := os.Getpid()
+
+	go func() {
```

Comment:
I haven't added a test for this, because there is a test in https://github.com/elastic/elastic-agent/pull/4822 that catches this 100% of the time.

Specifically it is this block of fixture_install.go:

```go
	f.t.Cleanup(func() {
		// check for running agents after uninstall had a chance to run
		processes := getElasticAgentProcesses(f.t)

		// there can be a single agent left when using --develop mode
		if f.installOpts != nil && f.installOpts.Develop {
			assert.LessOrEqual(f.t, len(processes), 1, "More than one agent left running at the end of the test when --develop was used: %v", processes)
			// The agent left running has to be the non-development agent. The development agent should be uninstalled first as a convention.
			if len(processes) > 0 {
				assert.NotContains(f.t, processes[0].Cmdline, paths.DevelopmentInstallDirName,
					"The agent installed with --develop was left running at the end of the test or was not uninstalled first: %v", processes)
			}
			return
		}

		assert.Empty(f.t, processes, "there should be no running agent at the end of the test")
	})
```

The `assert.LessOrEqual(f.t, len(processes), 1, ...` will fail without this change, because if there is an agent process running we alway pick up 2. One is the main process, and the other is the zombie created from the watcher. 

That test can't be extracted from that PR since it depends on the `--develop` option, and I didn't think it made sense to add another separate test, but let me know and I'll come up with something. Probably an integration test that calls `elastic-agent restart` and checks to make sure there's only one process running afterwards.